### ISSUE TITLE
MIOpen BatchNorm results in NaN in pyTorch

### ISSUE THREAD

**Issue by ashishfarmer** [ROCmSoftwarePlatform/MIOpen#248]
The issue is originally reported on ROCm PyTorch[ issue#657](https://github.com/ROCmSoftwarePlatform/pytorch/issues/657).

For the specific input tensor mentioned in the ticket, there are NaN's in the output. Verified that for the given test case, MIOpen BatchNorm is being used, and `MIOPEN_CHECK_NUMERICS` confirm the presence of NaNs in the output from MIOpen. 
```
MIOPEN_ENABLE_LOGGING_CMD=1 MIOPEN_CHECK_NUMERICS=1 python3.6 test.py
MIOpen(HIP): Info [Handle] stream: 0, device_id: 0
MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 64 -c 512 -H 1 -W 1 -m 0 --forw 1 -b 0 -r 1
MIOpen(HIP): Info [KernDb] database not present
MIOpen(HIP): Info [checkNumericsImpl] INPUT  ptr=0x7fbb90a00000 zeros=0 nans=0 infs=0  {64, 512, 1, 1}
MIOpen(HIP): Info [checkNumericsImpl] INPUT  ptr=0x7fbb90a20000 zeros=0 nans=0 infs=0  {1, 512, 1, 1}
MIOpen(HIP): Info [checkNumericsImpl] INPUT  ptr=0x7fbb90a20800 zeros=1 nans=0 infs=0  {1, 512, 1, 1}
MIOpen(HIP): Info [HipGetHccVersionImpl] HCC base: 3.1.20114
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv2_COv3, MIOpen version 2.5.0.8055-98bcbf4
MIOpen(HIP): Warning [checkNumericsImpl] OUTPUT ptr=0x7fbb90a21000 zeros=1 nans=1 infs=0  {64, 512, 1, 1}
Memory access fault by GPU node-2 (Agent handle: 0x3ef5350) on address (nil). Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

The tensor for which this behavior is seen is attached (in pytorch format) in the original ticket. The same data in numpy format to remove pytorch dependency is at [a.npy.zip](https://github.com/ROCmSoftwarePlatform/MIOpen/files/4698676/a.npy.zip)

cc: @sunway513

**Comment by aserio** [comment]
@ashishfarmer, What priority would you set for this issue?

**Comment by sunway513** [comment]
@aserio I'd suggest to mark it as P2.

**Comment by daniellowell** [comment]
@ashishfarmer If this is still an issue please feel free to reopen it.
